src/test/mesh_marker_test.cpp

```diff
@@ -38,26 +36,37 @@ publishText(int id, float x, float y, const std::string & text)
 
 void
 publishMesh(
-  int id, float x, float y,
+  int& id, float x, float y,
   float r, float g, float b, float a,
-  bool use_embedded_materials, bool mesh)
+  bool use_embedded_materials, int mesh)
 {
   using visualization_msgs::Marker;
 
   Marker marker;
   marker.header.frame_id = "/base_link";
   marker.header.stamp = ros::Time::now();
   marker.ns = "mesh";
-  if (mesh) {
-    marker.type = Marker::MESH_RESOURCE;
-  } else {
-    marker.type = Marker::SPHERE;
+  switch (mesh)
+  {
+    case 0:
+      marker.type = Marker::MESH_RESOURCE;
+      marker.mesh_resource = "package://rviz/src/test/meshes/pr2-base.dae";
+      marker.mesh_use_embedded_materials = use_embedded_materials;
+      break;
+    case 1:
+      marker.type = Marker::MESH_RESOURCE;
+      marker.mesh_resource = "package://rviz/src/test/meshes/frame.dae";
+      marker.mesh_use_embedded_materials = use_embedded_materials;
+      break;
+    case 2:
+      marker.type = Marker::SPHERE;
+      break;
   }
   marker.action = Marker::ADD;
   marker.pose.orientation.x = 0.0;
   marker.pose.orientation.y = 0.0;
-  marker.pose.orientation.z = 0.1;
-  marker.pose.orientation.w = 1.0;
+  marker.pose.orientation.z = 0.0995;
+  marker.pose.orientation.w = 0.995;
```

Comment:
What's the purpose of this?

Reply:
I suppose it doesn't matter since this is the test program, just curious.

Reply:
This is just normalizing the quaternion. Having some more restrictive checking in place (#1275),  we should provide correctly normalized quaternions...

Reply:
Ah makes sense. 👍

Reply:
Please don't yet merge this PR, even if you have approved. I want to wait for some feedback from affected people (https://github.com/ros-visualization/rviz/issues/1120#issuecomment-529695711) until the weekend at least.

Reply:
For sure, I'll let you take the reins on that. I also asked Ian to have a look, as he does most of the rendering related work in Gazebo.